### PR TITLE
Add api

### PR DESCRIPTION
**Pull request by damdafayton**
- Create a new game with the name of your choice by using the API.
- Implement the "Refresh" button (receiving data from the API and parsing the JSON).
- Implement the form "Submit" button (sending data to the API).
- Use async and await JavaScript features to consume the API.
- No styling is required.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,82 @@
...
+
+let gameID = 0;
+async function createGameId() {
+  const data = {
+    name: 'leader board',
+  };
+  let response = await fetch(`${leaderBoardApi.action}games/`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json', // 'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly remove commented code as they make your codebase look messy and hard to read.

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,82 @@
...
+    },
+    body: JSON.stringify(data),
+  });
+  response = await response.json(); // .then(reply => reply.json())
+  // .then(json =>
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly remove commented code.

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,82 @@
...
+
+async function createGame() {
+  const user = leaderBoardApi.elements.user.value;
+  const score = leaderBoardApi.elements.score.value;
+  const data = { user, score };
+
+  let saveGame = await fetch(`${leaderBoardApi.action}games/${gameID}/scores/`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json', // 'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (TedLivist) [inline src/index.js]: - [x] Remove commented code.

File: src/index.js [hunk 4]
```diff
@@ -1,4 +1,82 @@
 import './style.scss';
 
 // You can specify which plugins you need
-// import { Tooltip, Toast, Popover } from 'bootstrap';
\ No newline at end of file
+// import { Tooltip, Toast, Popover } from 'bootstrap';`
```
Comment (TedLivist) [inline src/index.js]: - [x] Remove commented code.

File: src/index.js [hunk 5]
```diff
@@ -1,4 +1,82 @@
...
+
+  const { result } = response;
+  const indexColumn = result.indexOf(':');
+  const indexAdded = result.indexOf('added');
+  console.log(result);
+  gameID = result.substring(indexColumn + 2, indexAdded - 1);
+  console.log(gameID);
+}
+
+async function createGame() {
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly refactor your function to use ES6 syntax (arrow function).

File: src/index.js [hunk 6]
```diff
@@ -1,4 +1,82 @@
...
+      const { score } = game;
+      const li = document.createElement('li');
+      li.innerText = `${user}: ${score}`;
+      scoreList.appendChild(li);
+    });
+  }
+}
+
+const leaderBoardSubmit = document.querySelector('#leaderboard-submit');
+async function submitHandler(e) {
```
Comment (TedLivist) [inline src/index.js]: - [ ] Kindly refactor to arrow function

File: src/index.js [hunk 7]
```diff
@@ -1,4 +1,82 @@
...
+    await createGameId();
+    createGame();
+  } else {
+    console.log('there is game id');
+    createGame();
+  }
+}
+leaderBoardSubmit.addEventListener('click', submitHandler);
+
+async function btnRefreshHandler() {
```
Comment (TedLivist) [inline src/index.js]: - [ ] Kindly refactor to arrow function

File: src/index.js [hunk 8]
```diff
@@ -1,4 +1,82 @@
...
 
 // You can specify which plugins you need
-// import { Tooltip, Toast, Popover } from 'bootstrap';
\ No newline at end of file
+// import { Tooltip, Toast, Popover } from 'bootstrap';`
+
+const leaderBoardApi = document.querySelector('#leaderboard-api');
+
+let gameID = 0;
+async function createGameId() {
```
Comment (TedLivist) [inline src/index.js]: - [ ] Kindly refactor to arrow function

File: src/index.js [hunk 9]
```diff
@@ -1,4 +1,82 @@
...
+    headers: {
+      'Content-Type': 'application/json', // 'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: JSON.stringify(data),
+  });
+  saveGame = await saveGame.json();
+  console.log(saveGame);
+}
+
+function populateScores(games) {
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly refactor to arrow function

File: src/index.js [hunk 10]
```diff
@@ -1,4 +1,82 @@
...
+  });
+  response = await response.json(); // .then(reply => reply.json())
+  // .then(json =>
+
+  const { result } = response;
+  const indexColumn = result.indexOf(':');
+  const indexAdded = result.indexOf('added');
+  console.log(result);
+  gameID = result.substring(indexColumn + 2, indexAdded - 1);
+  console.log(gameID);
```
Comment (TedLivist) [inline src/index.js]: - [x] Console logs kill performance and could expose confidential data in more sophisticated apps. Kindly remove them here.

File: src/index.js [hunk 11]
```diff
@@ -1,4 +1,82 @@
...
+
+  let saveGame = await fetch(`${leaderBoardApi.action}games/${gameID}/scores/`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json', // 'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: JSON.stringify(data),
+  });
+  saveGame = await saveGame.json();
+  console.log(saveGame);
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly remove console log

File: src/index.js [hunk 12]
```diff
@@ -1,4 +1,82 @@
...
+
+const leaderBoardSubmit = document.querySelector('#leaderboard-submit');
+async function submitHandler(e) {
+  e.preventDefault();
+
+  if (!gameID) {
+    await createGameId();
+    createGame();
+  } else {
+    console.log('there is game id');
```
Comment (TedLivist) [inline src/index.js]: - [x] Kindly remove console log

File: src/index.js [hunk 13]
```diff
@@ -1,4 +1,82 @@
...
+    createGame();
+  }
+}
+leaderBoardSubmit.addEventListener('click', submitHandler);
+
+async function btnRefreshHandler() {
+  let games = await fetch(`${leaderBoardApi.action}games/${gameID}/scores/`);
+  games = await games.json();
+  console.log(games);
```
Comment (TedLivist) [inline src/index.js]: - [x] Remove console log

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @damdafayton,

Good job so far!
✔️ Good documentation
✔️ Data is saved and retrieved from API
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 14]
```diff
@@ -1,4 +1,72 @@
...
+
+async function createGame() {
+  const user = leaderBoardApi.elements.user.value;
+  const score = leaderBoardApi.elements.score.value;
+  const data = { user, score };
+
+  await fetch(`${leaderBoardApi.action}games/${gameID}/scores/`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json', // 'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (ShoiraTa) [inline src/index.js]: - [ ] As requested by the previous reviewer, please remove commented code.

File: src/index.js [hunk 15]
```diff
@@ -1,4 +1,72 @@
...
+  });
+  response = await response.json();
+
+  const { result } = response;
+  const indexColumn = result.indexOf(':');
+  const indexAdded = result.indexOf('added');
+  gameID = result.substring(indexColumn + 2, indexAdded - 1);
+}
+
+async function createGame() {
```
Comment (ShoiraTa) [inline src/index.js]: - [ ] As requested by the previous reviewer, kindly refactor your function to use ES6 syntax (arrow function):
### Async arrow functions look like this:
```
const foo = async () => {
  // do something
}
```

File: src/index.js [hunk 16]
```diff
@@ -1,4 +1,72 @@
 import './style.scss';
 
-// You can specify which plugins you need
-// import { Tooltip, Toast, Popover } from 'bootstrap';
\ No newline at end of file
+const leaderBoardApi = document.querySelector('#leaderboard-api');
+
+let gameID = 0;
+async function createGameId() {
```
Comment (ShoiraTa) [inline src/index.js]: - [ ] As requested by the previous reviewer, kindly refactor your function to use ES6 syntax (arrow function):
### Async arrow functions look like this:
```
const foo = async () => {
  // do something
}
```

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @damdafayton ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by moise10r — APPROVED** [review]
### Hi @damdafayton , 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer 👏 
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] Kindly, Add a descriptive README file to your project, it will be great to add instructions to set the project up locally and instructions on how to run the project. This includes `npm install` , `npm start` etc

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 16 above)
Comment (moise10r) [reply src/index.js]: Well done!

**Pull request merged**